pkg/cmd/pr/create/create.go

```diff
@@ -341,7 +341,11 @@ func createRun(opts *CreateOptions) error {
 		Milestones: milestoneTitles,
 	}
 
+	doPreserveInput := false
+	defer shared.PreserveInput(opts.IO, &tb, defs, &doPreserveInput)()
```

Comment:
This may be too much magic but I was wondering if we could do something like https://play.golang.org/p/Fu9C96Gcr-C so that we don't need to tell the `PreserveInput` function when to preserve the input or not, it would be able to check if the calling function was going to return an error and then do its work if so.

Reply:
mm, i'm open to that. we don't tend to use named outputs but i think it could help, here